test/src/main/java/org/apache/accumulo/test/functional/HalfClosedTabletIT.java

```diff
@@ -0,0 +1,279 @@
+    // succeeds. We want to verify that the minc is being retried and the tablet
+    // can be closed.
+
+    try (AccumuloClient c = Accumulo.newClient().from(getClientProps()).build()) {
+
+      String tableName = getUniqueNames(1)[0];
+      final var tops = c.tableOperations();
+
+      tops.create(tableName);
+      final var tid = TableId.of(tops.tableIdMap().get(tableName));
+
+      try (BatchWriter bw = c.createBatchWriter(tableName)) {
+        Mutation m = new Mutation(new Text("r1"));
+        m.put("acf", tableName, "1");
+        bw.addMutation(m);
+      }
+
+      IteratorSetting setting = new IteratorSetting(50, "error", ErrorThrowingIterator.class);
+      setting.addOption(ErrorThrowingIterator.TIMES, "3");
+      c.tableOperations().attachIterator(tableName, setting, EnumSet.of(IteratorScope.minc));
+      c.tableOperations().compact(tableName, new CompactionConfig().setWait(true));
+
+      // Taking the table offline should succeed normally
+      tops.offline(tableName);
+
+      // Minc should have completed successfully
+      Wait.waitFor(() -> tabletHasExpectedRFiles(c, tableName, 1, 1, 1, 1), 340_000);
+
+      Wait.waitFor(() -> countHostedTablets(c, tid) == 0L, 340_000);
+
+    }
+  }
+
+  // Note that these tests can talk several minutes each because by the time the test
+  // code changes the configuration, the minc has failed so many times that the minc
+  // is waiting for a few minutes before trying again. For example, I saw this backoff
+  // timing:
+  //
+  // DEBUG: MinC failed sleeping 169 ms before retrying
+  // DEBUG: MinC failed sleeping 601 ms before retrying
+  // DEBUG: MinC failed sleeping 2004 ms before retrying
+  // DEBUG: MinC failed sleeping 11891 ms before retrying
+  // DEBUG: MinC failed sleeping 43156 ms before retrying
+  // DEBUG: MinC failed sleeping 179779 ms before retrying
+  @Test
+  public void testBadIteratorOnStack() throws Exception {
+
+    // In this scenario the table is using an iterator for minc that is throwing an exception.
+    // This test ensures that the Tablet can still be unloaded normally by taking if offline
+    // after the bad iterator has been removed from the minc configuration.
+
+    try (AccumuloClient c = Accumulo.newClient().from(getClientProps()).build()) {
+
+      String tableName = getUniqueNames(1)[0];
+      final var tops = c.tableOperations();
+
+      tops.create(tableName);
+      final var tid = TableId.of(tops.tableIdMap().get(tableName));
+
+      IteratorSetting is = new IteratorSetting(30, BadIterator.class);
+      c.tableOperations().attachIterator(tableName, is, EnumSet.of(IteratorScope.minc));
+
+      try (BatchWriter bw = c.createBatchWriter(tableName)) {
+        Mutation m = new Mutation(new Text("r1"));
+        m.put("acf", tableName, "1");
+        bw.addMutation(m);
+      }
+
+      c.tableOperations().flush(tableName, null, null, false);
+
+      UtilWaitThread.sleepUninterruptibly(30, TimeUnit.SECONDS);
```

Comment:
I think it would be ok to lower this time

```suggestion
      UtilWaitThread.sleepUninterruptibly(5, TimeUnit.SECONDS);
```

Reply:
I opted for a larger time because sometimes tests run at different speeds on different machines. Lowering could cause intermittent failures.

Reply:
Its not expected that the assertion after the sleep will fail no matter how long we sleep.